packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2314,6 +2314,31 @@ export class ContainerRuntime
 		return result;
 	}
 
+	/**
+	 * Returns an aliased data store with the given alias.
+	 * @param alias - The alias for the data store.
+	 * @returns - The aliased data store as a FluidObject. Throws usage error if the data store
+	 * is not aliased or if it doesn't exist.
+	 */
+	public async getAliasDataStore(alias: string): Promise<FluidObject> {
+		await this.dataStores.waitIfPendingAlias(alias);
+		const internalId = this.internalId(alias);
+		const context = await this.dataStores.getDataStore(internalId, { wait: false });
+		if (!(await context.isRoot())) {
+			throw new UsageError(`Requested data store ${alias} is not aliased`);
+		}
+		const channel = await context.realize();
+		if (channel.entryPoint !== undefined) {
+			return channel.entryPoint.get();
+		}
+
+		const response = await channel.request({ url: "/" });
+		if (response.status !== 200 || response.mimeType !== "fluid/object") {
+			throw new UsageError(`Request data store ${alias} does not exist`);
```

Comment:
Just to confirm, we're throwing when it doesn't exist in order to maintain the same behavior, right? Because if we don't (like with the `wait` param), why should we just return undefined?

Reply:
+1 for switching to undefined

Reply:
Yes, we can return undefined. I will do that.

Reply:
Changed it to return undefined. I exposed the `IResponseException` error thrown by `DataStores::getDataStore` function to check for the error code in case the data store doesn't exist.